From 8a0e53ea3880f1459166a8bb03085a8920edb058 Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Wed, 4 Sep 2013 16:33:43 +0200
Subject: [PATCH] fdo#68210: Make tiff import work again with merged-libs

- there was a symbol mismatch for dynamic loading
- while this fixes fdo#68210, similar fixes are likely needed for the other
  dynamically loaded functionality
- this also includes Change-Id: I6041ca10f58335187928995368db673c1ce22e7b

Change-Id: I7e4d6baf256dbe7b145dcbc118b4cac2075e07f5
---
 filter/source/graphicfilter/egif/egif.cxx       | 5 ++++-
 filter/source/graphicfilter/eos2met/eos2met.cxx | 5 ++++-
 filter/source/graphicfilter/epbm/epbm.cxx       | 5 ++++-
 filter/source/graphicfilter/epgm/epgm.cxx       | 5 ++++-
 filter/source/graphicfilter/epict/epict.cxx     | 5 ++++-
 filter/source/graphicfilter/eppm/eppm.cxx       | 5 ++++-
 filter/source/graphicfilter/eps/eps.cxx         | 5 ++++-
 filter/source/graphicfilter/eras/eras.cxx       | 5 ++++-
 filter/source/graphicfilter/etiff/etiff.cxx     | 9 ++++-----
 filter/source/graphicfilter/expm/expm.cxx       | 9 ++++-----
 filter/source/graphicfilter/idxf/idxf.cxx       | 5 ++++-
 filter/source/graphicfilter/ieps/ieps.cxx       | 5 ++++-
 filter/source/graphicfilter/ios2met/ios2met.cxx | 5 ++++-
 filter/source/graphicfilter/ipbm/ipbm.cxx       | 5 ++++-
 filter/source/graphicfilter/ipcd/ipcd.cxx       | 5 ++++-
 filter/source/graphicfilter/ipcx/ipcx.cxx       | 5 ++++-
 filter/source/graphicfilter/ipict/ipict.cxx     | 5 ++++-
 filter/source/graphicfilter/ipsd/ipsd.cxx       | 5 ++++-
 filter/source/graphicfilter/iras/iras.cxx       | 5 ++++-
 filter/source/graphicfilter/itga/itga.cxx       | 5 ++++-
 filter/source/graphicfilter/itiff/itiff.cxx     | 5 ++++-
 21 files changed, 84 insertions(+), 29 deletions(-)

diff --git a/filter/source/graphicfilter/egif/egif.cxx b/filter/source/graphicfilter/egif/egif.cxx
index 708e7b5..74b82a6 100644
--- a/filter/source/graphicfilter/egif/egif.cxx
+++ b/filter/source/graphicfilter/egif/egif.cxx
@@ -558,7 +558,10 @@ void GIFWriter::WriteTerminator()
 
 // ------------------------------------------------------------------------
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport egiGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/eos2met/eos2met.cxx b/filter/source/graphicfilter/eos2met/eos2met.cxx
index f5a5a23..73f074c 100644
--- a/filter/source/graphicfilter/eos2met/eos2met.cxx
+++ b/filter/source/graphicfilter/eos2met/eos2met.cxx
@@ -2552,7 +2552,10 @@ sal_Bool METWriter::WriteMET( const GDIMetaFile& rMTF, SvStream& rTargetStream,
 
 //================== GraphicExport - the exported Function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport emeGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/epbm/epbm.cxx b/filter/source/graphicfilter/epbm/epbm.cxx
index a32fd8a..4028ea9 100644
--- a/filter/source/graphicfilter/epbm/epbm.cxx
+++ b/filter/source/graphicfilter/epbm/epbm.cxx
@@ -189,7 +189,10 @@ void PBMWriter::ImplWriteNumber(sal_Int32 nNumber)
 // - exported function -
 // ---------------------
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport epbGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/epgm/epgm.cxx b/filter/source/graphicfilter/epgm/epgm.cxx
index ee91eb8..025e515 100644
--- a/filter/source/graphicfilter/epgm/epgm.cxx
+++ b/filter/source/graphicfilter/epgm/epgm.cxx
@@ -213,7 +213,10 @@ void PGMWriter::ImplWriteNumber(sal_Int32 nNumber)
 // - exported function -
 // ---------------------
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport epgGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/epict/epict.cxx b/filter/source/graphicfilter/epict/epict.cxx
index 8af1c52..60419f7 100644
--- a/filter/source/graphicfilter/epict/epict.cxx
+++ b/filter/source/graphicfilter/epict/epict.cxx
@@ -2266,7 +2266,10 @@ sal_Bool PictWriter::WritePict(const GDIMetaFile & rMTF, SvStream & rTargetStrea
 
 //================== GraphicExport - the exported Function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport eptGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/eppm/eppm.cxx b/filter/source/graphicfilter/eppm/eppm.cxx
index 666135a..84388f7 100644
--- a/filter/source/graphicfilter/eppm/eppm.cxx
+++ b/filter/source/graphicfilter/eppm/eppm.cxx
@@ -222,7 +222,10 @@ void PPMWriter::ImplWriteNumber(sal_Int32 nNumber)
 // - exported function -
 // ---------------------
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport eppGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/eps/eps.cxx b/filter/source/graphicfilter/eps/eps.cxx
index 454d683..7627a17 100644
--- a/filter/source/graphicfilter/eps/eps.cxx
+++ b/filter/source/graphicfilter/eps/eps.cxx
@@ -2752,7 +2752,10 @@ sal_Bool PSWriter::ImplGetBoundingBox( double* nNumb, sal_uInt8* pSource, sal_uL
 
 //================== GraphicExport - die exportierte Funktion ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport epsGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/eras/eras.cxx b/filter/source/graphicfilter/eras/eras.cxx
index d0f3eff..d40ee67 100644
--- a/filter/source/graphicfilter/eras/eras.cxx
+++ b/filter/source/graphicfilter/eras/eras.cxx
@@ -269,7 +269,10 @@ void RASWriter::ImplPutByte( sal_uInt8 nPutThis )
 // - exported function -
 // ---------------------
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport eraGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/etiff/etiff.cxx b/filter/source/graphicfilter/etiff/etiff.cxx
index b7ae837..68af37d 100644
--- a/filter/source/graphicfilter/etiff/etiff.cxx
+++ b/filter/source/graphicfilter/etiff/etiff.cxx
@@ -592,11 +592,10 @@ void TIFFWriter::EndCompression()
 
 // ------------------------------------------------------------------------
 
-// ---------------------
-// - exported function -
-// ---------------------
-
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport etiGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/expm/expm.cxx b/filter/source/graphicfilter/expm/expm.cxx
index 451ad2e..ad0e704 100644
--- a/filter/source/graphicfilter/expm/expm.cxx
+++ b/filter/source/graphicfilter/expm/expm.cxx
@@ -247,11 +247,10 @@ void XPMWriter::ImplWriteColor( sal_uInt16 nNumber )
 
 // ------------------------------------------------------------------------
 
-// ---------------------
-// - exported function -
-// ---------------------
-
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicExport expGraphicExport
 #endif
 
diff --git a/filter/source/graphicfilter/idxf/idxf.cxx b/filter/source/graphicfilter/idxf/idxf.cxx
index b2639b3..1a42be5 100644
--- a/filter/source/graphicfilter/idxf/idxf.cxx
+++ b/filter/source/graphicfilter/idxf/idxf.cxx
@@ -29,7 +29,10 @@ class FilterConfigItem;
 
 //================== GraphicImport - die exportierte Funktion ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport idxGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/ieps/ieps.cxx b/filter/source/graphicfilter/ieps/ieps.cxx
index e3bf1173..a4ddc18 100644
--- a/filter/source/graphicfilter/ieps/ieps.cxx
+++ b/filter/source/graphicfilter/ieps/ieps.cxx
@@ -468,7 +468,10 @@ void MakePreview(sal_uInt8* pBuf, sal_uInt32 nBytesRead,
 
 //================== GraphicImport - the exported function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport ipsGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/ios2met/ios2met.cxx b/filter/source/graphicfilter/ios2met/ios2met.cxx
index 834bd11..debf296 100644
--- a/filter/source/graphicfilter/ios2met/ios2met.cxx
+++ b/filter/source/graphicfilter/ios2met/ios2met.cxx
@@ -2704,7 +2704,10 @@ void OS2METReader::ReadOS2MET( SvStream & rStreamOS2MET, GDIMetaFile & rGDIMetaF
 
 //================== GraphicImport - the exported function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport imeGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/ipbm/ipbm.cxx b/filter/source/graphicfilter/ipbm/ipbm.cxx
index 5ca1c7b..4de5cef 100644
--- a/filter/source/graphicfilter/ipbm/ipbm.cxx
+++ b/filter/source/graphicfilter/ipbm/ipbm.cxx
@@ -520,7 +520,10 @@ sal_Bool PBMReader::ImplReadBody()
 
 //================== GraphicImport - the exported function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport ipbGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/ipcd/ipcd.cxx b/filter/source/graphicfilter/ipcd/ipcd.cxx
index d9f249b..57178d8 100644
--- a/filter/source/graphicfilter/ipcd/ipcd.cxx
+++ b/filter/source/graphicfilter/ipcd/ipcd.cxx
@@ -367,7 +367,10 @@ void PCDReader::ReadImage(sal_uLong nMinPercent, sal_uLong nMaxPercent)
 
 //================== GraphicImport - the exported Function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport icdGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/ipcx/ipcx.cxx b/filter/source/graphicfilter/ipcx/ipcx.cxx
index c45f98c..3738180 100644
--- a/filter/source/graphicfilter/ipcx/ipcx.cxx
+++ b/filter/source/graphicfilter/ipcx/ipcx.cxx
@@ -396,7 +396,10 @@ void PCXReader::ImplReadPalette( sal_uLong nCol )
 
 //================== GraphicImport - the exported function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport ipxGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/ipict/ipict.cxx b/filter/source/graphicfilter/ipict/ipict.cxx
index 49d9f92..9acd4c9 100644
--- a/filter/source/graphicfilter/ipict/ipict.cxx
+++ b/filter/source/graphicfilter/ipict/ipict.cxx
@@ -1864,7 +1864,10 @@ void PictReader::ReadPict( SvStream & rStreamPict, GDIMetaFile & rGDIMetaFile )
 
 //================== GraphicImport - the exported function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport iptGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/ipsd/ipsd.cxx b/filter/source/graphicfilter/ipsd/ipsd.cxx
index b693c13..4eccc13 100644
--- a/filter/source/graphicfilter/ipsd/ipsd.cxx
+++ b/filter/source/graphicfilter/ipsd/ipsd.cxx
@@ -720,7 +720,10 @@ sal_Bool PSDReader::ImplReadBody()
 
 //================== GraphicImport - the exported function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport ipdGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/iras/iras.cxx b/filter/source/graphicfilter/iras/iras.cxx
index 36241e5..8add8e8 100644
--- a/filter/source/graphicfilter/iras/iras.cxx
+++ b/filter/source/graphicfilter/iras/iras.cxx
@@ -337,7 +337,10 @@ sal_uInt8 RASReader::ImplGetByte()
 
 //================== GraphicImport - die exportierte Funktion ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport iraGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/itga/itga.cxx b/filter/source/graphicfilter/itga/itga.cxx
index e3967e9..06220e3 100644
--- a/filter/source/graphicfilter/itga/itga.cxx
+++ b/filter/source/graphicfilter/itga/itga.cxx
@@ -797,7 +797,10 @@ sal_Bool TGAReader::ImplReadPalette()
 
 //================== GraphicImport - die exportierte Funktion ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport itgGraphicImport
 #endif
 
diff --git a/filter/source/graphicfilter/itiff/itiff.cxx b/filter/source/graphicfilter/itiff/itiff.cxx
index 7b47690..71bd7e0 100644
--- a/filter/source/graphicfilter/itiff/itiff.cxx
+++ b/filter/source/graphicfilter/itiff/itiff.cxx
@@ -1358,7 +1358,10 @@ sal_Bool TIFFReader::ReadTIFF(SvStream & rTIFF, Graphic & rGraphic )
 
 //================== GraphicImport - the exported function ================
 
-#if defined(DISABLE_DYNLOADING) || defined(LIBO_MERGELIBS)
+// this needs to be kept in sync with
+// ImpFilterLibCacheEntry::GetImportFunction() from
+// vcl/source/filter/graphicfilter.cxx
+#if defined(DISABLE_DYNLOADING)
 #define GraphicImport itiGraphicImport
 #endif
 
-- 
1.8.3.2

